flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/runtime/batch/sql/join/LookupJoinITCase.scala

```diff
@@ -301,28 +374,46 @@ class LookupJoinITCase(legacyTableSource: Boolean, isAsyncMode: Boolean, cacheTy
       BatchTestBase.row(null, 11, null),
       BatchTestBase.row(null, 12, null))
     checkResult(sql, expected)
+    afterEach(legacyTableSource)
   }
 
-  @Test
-  def testJoinTemporalTableOnNullConstantKey(): Unit = {
+  @ParameterizedTest
+  @MethodSource(Array("parameters"))
+  def testJoinTemporalTableOnNullConstantKey(
+      legacyTableSource: Boolean,
+      isAsyncMode: Boolean,
+      cacheType: LookupCacheType): Unit = {
+    beforeEach(legacyTableSource, isAsyncMode, cacheType);
     val sql = s"SELECT T.id, T.len, T.content FROM T JOIN userTable " +
       "for system_time as of T.proctime AS D ON D.id = null"
     val expected = Seq()
     checkResult(sql, expected)
+    afterEach(legacyTableSource)
   }
 
-  @Test
-  def testJoinTemporalTableOnMultiKeyFieldsWithNullConstantKey(): Unit = {
+  @ParameterizedTest
+  @MethodSource(Array("parameters"))
+  def testJoinTemporalTableOnMultiKeyFieldsWithNullConstantKey(
+      legacyTableSource: Boolean,
+      isAsyncMode: Boolean,
+      cacheType: LookupCacheType): Unit = {
+    beforeEach(legacyTableSource, isAsyncMode, cacheType);
     val sql = s"SELECT T.id, T.len, D.name FROM T JOIN userTable " +
       "for system_time as of T.proctime AS D ON T.content = D.name AND null = D.id"
     val expected = Seq()
     checkResult(sql, expected)
+    afterEach(legacyTableSource)
   }
 
-  @Test
-  def testJoinTemporalTableWithComputedColumn(): Unit = {
+  @ParameterizedTest
+  @MethodSource(Array("parameters"))
+  def testJoinTemporalTableWithComputedColumn(
+      legacyTableSource: Boolean,
+      isAsyncMode: Boolean,
+      cacheType: LookupCacheType): Unit = {
+    beforeEach(legacyTableSource, isAsyncMode, cacheType);
     // Computed column do not support in legacyTableSource.
-    Assume.assumeFalse(legacyTableSource)
+    assertThat(legacyTableSource).isFalse
```

Comment:
```suggestion
    assumeThat(legacyTableSource).isFalse
```
since it is an assumption it should probably be `assumeThat`

Reply:
Done.